daffodil-lib/src/main/scala/org/apache/daffodil/util/DefaultValidator.scala

```diff
@@ -0,0 +1,117 @@
+
+/**
+ * Use this for extra validation passes in the TDML Runner
+ * to do a validation pass on the TDML expected Infoset w.r.t. the model and to
+ * do a validation pass on the actual result w.r.t. the model as an XML document.
+ */
+class DefaultValidator(schemaFileNames: Seq[String])
+  extends Validator
+  with ErrorHandler {
+
+  private var errors = List.empty[ValidationFailure]
+  private var warnings = List.empty[ValidationWarning]
+
+  private val schemaSources: Seq[javax.xml.transform.Source] = schemaFileNames.map { fn =>
+    val uri = new URI(fn)
+    val is = uri.toURL.openStream()
+    val stream = new StreamSource(is)
+    stream.setSystemId(uri.toString) // must set this so that relative URIs will be created for import/include files.
+    stream
+  }
+
+  private val factory = new org.apache.xerces.jaxp.validation.XMLSchemaFactory()
+  private val resolver = DFDLCatalogResolver.get
+  factory.setResourceResolver(resolver)
+
+  private val schema = factory.newSchema(schemaSources.toArray)
+
+  private val validator = new ThreadLocal[XercesValidator] {
+    override def initialValue(): XercesValidator =
+      initializeValidator(schema.newValidator(), DefaultValidator.this, resolver)
+  }
+
+  def validateXML(document: java.io.InputStream): ValidationResult = {
+    val documentSource = new StreamSource(document)
+
+    try validator.get().validate(documentSource)
+    catch {
+      // Some SAX Parse errors are thrown even if you specify an error handler to the
+      // validator. So we also need this catch
+      case e: SAXException =>
+        errors :+= SaxValidationError(e)
+        ValidationResult(warnings, errors)
+    }
+    ValidationResult(warnings, errors)
+  }
+
+  def initializeValidator(validator: javax.xml.validation.Validator, errHandler: ErrorHandler, resolver: DFDLCatalogResolver):
+    javax.xml.validation.Validator = {
+    validator.setFeature(XMLConstants.FEATURE_SECURE_PROCESSING, true)
+    //
+    validator.setFeature("http://xml.org/sax/features/validation", true)
+
+    // If you enable the feature below, it seems to do no validation at all. Just passes.
+    //          validator.setFeature("http://apache.org/xml/features/validation/dynamic", true)
+
+    validator.setFeature("http://apache.org/xml/features/validation/schema", true)
+    validator.setFeature("http://apache.org/xml/features/validation/schema-full-checking", true)
+    validator.setErrorHandler(errHandler)
+    validator.setResourceResolver(resolver)
+    validator
+  }
+
+  override def warning(spe: SAXParseException): Unit =  warnings :+= SaxValidationWarning(spe)
+  override def error(spe: SAXParseException): Unit = errors :+= SaxValidationError(spe)
+  override def fatalError(spe: SAXParseException): Unit = errors :+= SaxValidationError(spe)
+}
+
+object DefaultValidator {
+  type XercesValidator = javax.xml.validation.Validator
+}
+
+class DefaultValidatorFactory extends ValidatorFactory {
```

Comment:
This, could stay DefaultValidatorFactory that happens to create a XercesValidator. 
Or it could change to be XercesValidatorFactory.